core/types/calls.cc

```diff
@@ -2451,6 +2451,140 @@ class Magic_selfNew : public IntrinsicMethod {
     }
 } Magic_selfNew;
 
+class Magic_checkAndAnd : public IntrinsicMethod {
+public:
+    void apply(const GlobalState &gs, const DispatchArgs &args, DispatchResult &res) const override {
+        // args[0] is recv
+        // args[1] is the && tmp var
+        // args[2] is the method name
+        // args[3...] are the args
```

Comment:
A high-level comment on what we expect the desugarer to produce as it relates to the original `lhs && rhs.foo`, and how we consume that info, might be nice here.

Reply:
hm, isn't that what this comment is? i'm not sure what more you'd like

Reply:
Ah gotcha. I would have assumed that to be implied from the fact that this is an intrinsic for a `<Magic>` method

Reply:
I guess my basic point is that the purpose of the `<Magic>` method itself isn't fully clear unless you go searching for the code that emits it, so documenting that would be good. (That applies to a lot of the `<Magic>` stuff, though!)

Reply:
(i added a comment)

Reply:
Thanks! That's exactly what I was looking for.